test/api.js

```diff
@@ -616,6 +616,26 @@ test('test file with only skipped tests does not create a failure', function (t)
 		});
 });
 
+test('test file with exclusive tests causes non-exclusive tests in other files to be ignored', function (t) {
+	t.plan(4);
+
+	var files = [
+		path.join(__dirname, 'fixture/exclusive.js'),
+		path.join(__dirname, 'fixture/exclusive-nonexclusive.js'),
+		path.join(__dirname, 'fixture/one-pass-one-fail.js')
+	];
+
+	var api = new Api(files);
```

Comment:
`master` moves the `files` argument to `api.run(files)`. You'll have to rebase this branch, sorry :)